Signal/src/call/CallService.swift

```diff
@@ -1103,33 +1103,34 @@ protocol CallServiceObserver: class {
     func setHasLocalVideo(hasLocalVideo: Bool) {
         AssertIsOnMainThread()
 
-        let authStatus = AVCaptureDevice.authorizationStatus(forMediaType:AVMediaTypeVideo)
-        switch authStatus {
-        case .notDetermined:
-            Logger.debug("\(TAG) authStatus: AVAuthorizationStatusNotDetermined")
-            break
-        case .restricted:
-            Logger.debug("\(TAG) authStatus: AVAuthorizationStatusRestricted")
-            break
-        case .denied:
-            Logger.debug("\(TAG) authStatus: AVAuthorizationStatusDenied")
-            break
-        case .authorized:
-            Logger.debug("\(TAG) authStatus: AVAuthorizationStatusAuthorized")
-            break
-        }
-
-        // We don't need to worry about the user granting or remoting this permission
-        // during a call while the app is in the background, because changing this
-        // permission kills the app.
-        if authStatus != .authorized {
-
-            OWSAlerts.showAlert(withTitle:NSLocalizedString("MISSING_CAMERA_PERMISSION_TITLE", comment: "Alert title when camera is not authorized"),
-                                message:NSLocalizedString("MISSING_CAMERA_PERMISSION_MESSAGE", comment: "Alert body when camera is not authorized"))
-
+        guard let frontmostViewController = UIApplication.shared.frontmostViewController else {
```

Comment:
Never ask for camera or microphone permissions if app is not active.  I think I'm going to modify this to only skip if in background.  iOS could temporarily make app inactive to show a system alert, e.g. "iOS update available."